plugins/module_utils/network/ios/rm_templates/snmp_server.py

```diff
@@ -0,0 +1,1720 @@
+)
+
+
+def cmd_option_engine_id(config_data):
+    cmd = ""
+    if config_data:
+        cmd = "snmp-server engineID "
+        if config_data.get("local"):
+            cmd += "local"
+        if config_data.get("remote"):
+            rm = config_data.get("remote")
+            if rm.get("host"):
+                cmd += "remote {host}".format(host=rm.get("host"))
+            if rm.get("udp_port"):
+                cmd += " udp-port {udp_port}".format(
+                    udp_port=rm.get("udp_port")
+                )
+            if rm.get("vrf"):
+                cmd += " vrf {vrf}".format(vrf=rm.get("vrf"))
+        if config_data.get("id"):
+            cmd += " {id}".format(id=config_data.get("id"))
+    return cmd
+
+
+def cmd_option_file_transfer(config_data):  # contain sub list attr
+    cmd = ""
+    if config_data.get("file_transfer"):
+        conf = config_data.get("file_transfer")
+        cmd = "snmp-server file-transfer"
+        if conf.get("access_group"):
+            cmd += " access-group {ag}".format(ag=conf.get("access_group"))
+        if conf.get("protocol"):
+            cmd += " protocol"
+            for protocol in list(conf.get("protocol").keys()):
+                cmd += " {protocol}".format(protocol=protocol)
+    return cmd
+
+
+def cmd_option_hosts(config_data):  # contain sub list attr
+    cmd = ""
+    if config_data:
+        cmd = "snmp-server host"
+        if config_data.get("host"):
+            cmd += " {host}".format(host=config_data.get("host"))
+        if config_data.get("informs"):
+            cmd += " informs"
+        if config_data.get("version"):
+            cmd += " version {version}".format(
+                version=config_data.get("version")
+            )
+        if config_data.get("version_option"):
+            cmd += " {version}".format(
+                version=config_data.get("version_option")
+            )
+        if config_data.get("vrf"):
+            cmd += " vrf {vrf}".format(vrf=config_data.get("vrf"))
+        if config_data.get("community_string"):
+            cmd += " {community_string}".format(
+                community_string=config_data.get("community_string")
+            )
+        if config_data.get("traps"):
+            for protocol in list(config_data.get("traps").keys()):
+                cmd += " {protocol}".format(protocol=protocol)
+    return cmd
+
+
+def cmd_option_trap_bgp(config_data):
+    cmd = ""
+    conf = config_data.get("traps", {}).get("bgp", {})
+    if conf:
+        if conf.get("enable"):
+            cmd += "snmp-server enable traps bgp"
+        if conf.get("cbgp2"):
+            cmd += " cbgp2"
+        if conf.get("state_changes"):
+            if conf.get("state_changes").get("enable"):
+                cmd += " state-changes"
+            if conf.get("state_changes").get("all"):
+                cmd += " all"
+            if conf.get("state_changes").get("backward_trans"):
+                cmd += " backward-trans"
```

Comment:
is state-changes missing in cmd formation?

Reply:
it is added as per the expected command formation in ln 95